scene/gui/code_edit.cpp

```diff
@@ -2959,9 +2959,16 @@ void CodeEdit::_filter_code_completion_candidates_impl() {
 		}
 
 		if (in_string != -1) {
+			// The completion string may have a literal behind it, which should be removed before re-quoting.
+			String literal = "";
+			if (option.insert_text.right(-1).is_quoted()) {
```

Comment:
```suggestion
			if (option.insert_text.substr(1).is_quoted()) {
```
Less for all these

Reply:
Not equivalent. right(-1) with negative is like trim_left(1). substr(1) is like left(1).

Reply:
No `right(-1)` is "everything except first character" no? `substr(1)` is not `left(1)`, `left(1)` is `substr(0, 1)`

Reply:
Oh, true, my bad, I thought I remembered this right but I didn't.

Reply:
I believe it changed from 3.x and it's confusing